src/inc/volatile.h

```diff
@@ -120,7 +120,7 @@ T VolatileLoad(T const * pt)
     STATIC_CONTRACT_SUPPORTS_DAC_HOST_ONLY;
 
 #ifndef DACCESS_COMPILE
-#if defined(_ARM64_) && defined(__GNUC__)
+#if defined(_ARM64_) && defined(__GNUC__) && !(defined(__ANDROID__))
```

Comment:
Would `defined(__clang__) && __has_builtin(__atomic_load)` work? We use `__has_builtin` in other places.